modules/market/src/main/java/com/opengamma/strata/market/interpolator/DoubleQuadraticCurveInterpolator.java

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.interpolator;
+
+import java.io.Serializable;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.math.impl.function.RealPolynomialFunction1D;
+import com.opengamma.strata.math.impl.interpolation.WeightingFunction;
+import com.opengamma.strata.math.impl.interpolation.WeightingFunctions;
+
+/**
+ * Interpolator implementation that uses double quadratic interpolation.
+ * <p>
+ * This uses linear weighting.
+ */
+final class DoubleQuadraticCurveInterpolator
+    implements CurveInterpolator, Serializable {
+
+  /**
+   * The serialization version id.
+   */
+  private static final long serialVersionUID = 1L;
+  /**
+   * The interpolator name.
+   */
+  public static final String NAME = "DoubleQuadratic";
+  /**
+   * The interpolator instance.
+   */
+  public static final CurveInterpolator INSTANCE = new DoubleQuadraticCurveInterpolator();
+  /**
+   * The weighting function.
+   */
+  private static final WeightingFunction WEIGHT_FUNCTION = WeightingFunctions.LINEAR;
+
+  /**
+   * Restricted constructor.
+   */
+  private DoubleQuadraticCurveInterpolator() {
+  }
+
+  // resolve instance
+  private Object readResolve() {
+    return INSTANCE;
+  }
+
+  //-------------------------------------------------------------------------
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public BoundCurveInterpolator bind(DoubleArray xValues, DoubleArray yValues) {
+    return new Bound(xValues, yValues);
+  }
+
+  //-----------------------------------------------------------------------
+  @Override
+  public String toString() {
+    return NAME;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Bound interpolator.
+   */
+  static class Bound extends AbstractBoundCurveInterpolator {
+    private final double[] xValues;
+    private final double[] yValues;
+    private final int intervalCount;
+    private final RealPolynomialFunction1D[] quadratics;
+    private final RealPolynomialFunction1D[] quadraticsFirstDerivative;
+
+    Bound(DoubleArray xValues, DoubleArray yValues) {
+      super(xValues, yValues);
+      this.xValues = xValues.toArrayUnsafe();
```

Comment:
Are you sure you want `toArrayUnsafe` here? They are passed in from the `bind` method and come from the outside world without any cloning.